### PR TITLE
Make puku rewrite build files by default, with a flag to skip rewriting

### PR DESCRIPTION
**Pull request by toastwaffle**
This contains an unfortunate amount of plumbing to pass the `skipRewriting` flag around, but short of a fairly violent refactoring of the command structure I'm not sure we can do anything better.

The net result of this is that BUILD files output by puku without the `--skip_rewriting` flag should satisfy `plz fmt`

I tested this on the puku repository itself - With the flag, a number of BUILD files were updated with semantic changes only; without it the same set of files were updated, but with some additional rewriting changes too. No files were changed that didn't have semantic changes. ~I'll do a separate PR after this one to `plz fmt` and `plz puku fmt` all the BUILD files~ (already done in #122 so I could cleanly update deps in this PR)

### CODE REVIEW

File: graph/graph.go [hunk 1]
```diff
@@ -231,11 +233,46 @@ func checkVisibility(target labels.Label, visibilities []string) bool {
...
+	return outputFormattedBuildFile(buildFile, outFn, format, skipRewriting)
+}
+
+// saveFormattedBuildFile writes a build file to disk if puku has made meaningful changes.
+//
+// See the comment on outputFormattedBuildFile for more details.
+func saveFormattedBuildFile(buildFile *build.File, skipRewriting bool) error {
+	outFn := func() (io.WriteCloser, error) {
+		return os.Create(buildFile.Path)
+	}
```
Comment (Tatskaari) [inline graph/graph.go:258]: Found this a bit hard to follow. Can't you just create the file here and pass in the writer to `outputFormattedBuildFile()` rather than using a closure? This function can defer the close.

Comment (toastwaffle) [reply graph/graph.go:258]: I've added a comment to explain why it's like this.

**Review by Tatskaari — APPROVED** [review]
